### ISSUE TITLE
Contact Source. How to pass source or tag variable in URL?

### ISSUE THREAD

**Issue by Strellman** [DiscipleTools/disciple-tools-webform#52]
It would be really helpful to know how somebody got to our website! We are storing a cookie to indicate which ad, or which person led the contact to us. Now I want to pass that field to DT in the URL so that as the contact submits their form, the dispatcher will know where they came from (possibly a city, a team, or a person).
If someone can at least point me to the section of code, or the variables to set, that would even help. I would like to create a snippet to store this info.

Another approach would be to do the form on the host website and just send the fields to DT somehow.

**Comment by Strellman** [comment]
WPFORMS has done a lot of this. See https://wpforms.com/wpforms-hidden-field/
Is there anyway to to pass the WPFORM data to DT?

**Comment by corsacca** [comment]
> It would be really helpful to know how somebody got to our website!

Agreed. The current webform does not really support a way of doing that.
One solution could be to add a hook to https://github.com/DiscipleTools/disciple-tools-webform/blob/e38458afa94819c97e2a4c4091ef228d153218f1/includes/create-contact.php#L75 and hook into it with some php code from your server.

Or, you could write your own form and send the fields to D.T. This is possible via the [API](https://developers.disciple.tools/theme-core/api-posts/create-post) and the [Site to Site like authentication](https://developers.disciple.tools/theme-core/authentication/site-to-site-link)

The webform could support hidden fields. Since we display the form via iframe, updating those fields from the website is challenging. [Next upgrade](https://github.com/DiscipleTools/disciple-tools-webform/issues/49)

**Comment by Strellman** [comment]
Thanks Corsacca, that is very helpful. 
Cookies can't be pulled from a different site into DT, but maybe they could be passed as arguments.  What about getting the cookie before generating the iframe and passing it along with referer in the iframe url as a variable? Hopefully that will not be the "Contact us" url.

Here is what I am thinking. Tell me if it makes sense to you. 
`//send DT tag of referer at least
$tags = 'tags=':$_SERVER['HTTP_REFERER'];
//check to see if user has our cookie for tag
$tag = $_COOKIE['tag'];
if(isset($tag) 
$tags := '&':$tag;

$iframestring = '<iframe src="https://example.com/wp-content/plugins/disciple-tools-webform-master/public/form.php?tags=reserved;token=c234234234234480"'; style="width:100%;height:550px;" frameborder="0"></iframe>';
$iframestring = str_replace('reserved', $tags, $iframestring);
echo $iframestring;
return;`

Then on the DT instance (not sure where to put this if I add the Code Snippets plugin).
`//Get tags into DT contact
function dt_post_create_fields( array $fields, $string post_type);
//check if we are dealing with a contact
if ($post_type === "contacts"){
//check if the tags field is already set
if ( !isset( $fields["tags"] )){
$tags = sanitize_text_field($_GET['tags'] ); 
if isset($tags)  {
//define the tags field
$fields["tags"] = $tags
}
}
}
return $fields;
`

**Comment by Strellman** [comment]
This snippet is a shortcode to insert our reserved_tags variables into the iframe before it is called. Usage: 
[dt_source]<iframe src="https://example.com/wp-content/plugins/disciple-tools-webform-master/public/form.php?tags=reserved_tags;token=c234234234234480"'; style="width:100%;height:550px;" frameborder="0"></iframe>[/dt_source]

//register shortcode
add_shortcode('dt_source','dt_source');

function dt_source($attr, $iframestring = null){

$tags = 'tags=';
//check to see if user has our cookie
$contact_source = $_COOKIE['contact_source'];
if (isset($contact_source) $tags := $contact_source.'&'
 
//send DT tag of referer at least
$tags .= $_SERVER['HTTP_REFERER'];

//replace reserved_tags in $iframestring 
$iframestring = str_replace('reserved_tags', $tags, $iframestring);
return $iframestring;

**Comment by corsacca** [comment]
That could possibly work
Step 1 pass the parameters to the iframe
Step 2 read the url parameters via javascript and submit the form with those parameters. 
Step 3 Make sure those parameters are passed along in the form_submit() function

The campaigns field (acts like the tags field) might be the right field for this data. And could be supported out of the box.
If the data is different for each contact, then a text field would be better. And step 3 would be more complicated

We'd want to use a field that already exists in D.T or set up a custom field, so we won't need to do anything in `dt_post_create_fields`. and instead make sure `form_submit()` in the webform plugin gets and sends the right data
